fedora/client/bodhi.py

```diff
@@ -24,22 +24,376 @@
 .. moduleauthor:: Toshio Kuratomi <tkuratom@redhat.com>
 """
 
+import os
 import logging
+import textwrap
+import warnings
+import requests
 
-from textwrap import wrap
-from os.path import join, expanduser, exists
+from fedora.client import OpenIdBaseClient, FedoraClientError, BaseClient
+import fedora.client.openidproxyclient
 
-from fedora.client import BaseClient, FedoraClientError
-
-__version__ = '0.5.1'
+__version__ = '2.0.0'
 log = logging.getLogger(__name__)
 
+BASE_URL = 'https://admin.fedoraproject.org/updates/'
+STG_BASE_URL = 'https://admin.stg.fedoraproject.org/updates/'
+STG_OPENID_API = 'https://id.stg.fedoraproject.org/api/v1/'
+
 
 class BodhiClientException(FedoraClientError):
     pass
 
 
-class BodhiClient(BaseClient):
+def BodhiClient(base_url=BASE_URL, staging=False, **kwargs):
+    if staging:
+        log.info('Using bodhi2 STAGING environment')
+        base_url = STG_BASE_URL
+        fedora.client.openidproxyclient.FEDORA_OPENID_API = STG_OPENID_API
+
+    log.debug('Querying bodhi API version')
+    api_url = base_url + 'api_version'
+    response = requests.head(api_url)
+    if response.status_code == 200:
+        print('Bodhi 2 detected')
```

Comment:
Do we want to remove these print statements before we ship this?  (or demote them to log.debug statements?)